packages/webpack-loader/index.js

```diff
@@ -20,28 +22,33 @@ function localeFromResourcePath(resourcePath, locales) {
 }
 
 module.exports = function loadMessages(content) {
-  var messages = YAML.parse(content);
-  var options = loaderUtils.getOptions(this) || {};
-  var locale = options.locale;
-  if (options.convert) {
-    var cm = convert(messages, options.convert);
-    if (!locale) {
-      locale = cm.locales;
-    }
+  let { convert, encoding, locale, propKeyPath = true, ...mfOpt } =
+    loaderUtils.getOptions(this) || {};
+
+  if (!encoding || encoding === 'auto')
+    encoding = uv(content) ? 'utf8' : 'latin1';
+  const input = content.toString(encoding);
+
+  let messages = this.resourcePath.endsWith('.properties')
+    ? parse(input, propKeyPath)
+    : YAML.parse(input);
+
+  if (convert) {
+    const cm = convertToMessageFormat(messages, convert);
+    if (!locale) locale = cm.locales;
     messages = cm.translations;
   }
+
   if (typeof locale === 'string' && locale.indexOf(',') !== -1)
     locale = locale.split(',');
   if (Array.isArray(locale) && locale.length > 1) {
-    var relPath = path.relative(process.cwd(), this.resourcePath);
+    const relPath = relative(process.cwd(), this.resourcePath);
     locale = localeFromResourcePath(relPath, locale);
   }
-  const mfOpt = {};
-  if (options.biDiSupport) mfOpt.biDiSupport = true;
-  if (options.customFormatters || options.formatters) {
-    mfOpt.customFormatters = options.customFormatters || options.formatters;
```

Comment:
Does messageformat/core respect `formatters` now? Otherwise this will be a breaking change, albeit in an undocumented feature since we never listed `formatters` in the [README](https://github.com/messageformat/messageformat/tree/df713b1714837b92e2bf4c07d5323c21c78df278/packages/webpack-loader#options).

Reply:
You're right, but it should be fine in this case. Due to the overall v3 update it seems like a good time to finally release this as a 1.0.0, and with #290 the package is even getting renamed to `@messageformat/loader`. So it's a good time for nominal breaking changes.

The remaining question is whether we need to explicitly mention the deprecation of an undocumented feature in the release notes, and I'd suggest that we're probably ok not doing so.

Reply:
Agreed, no doc should be fine especially with a major version bump. Kiitos!